### ISSUE TITLE
What is the unit of the minimum-font-size option?

### ISSUE THREAD

**Issue by zardilior** [wkhtmltopdf/wkhtmltopdf#4234]
It is not clear, it is an int, but the unit for the result is not clear nor if this moves between os

**Comment by nwhitmont** [comment]
I would also like to know what units are the default here!

mm/cm/em/px?

**Comment by zardilior** [comment]
Still no answer?

**Comment by PhilterPaper** [comment]
Without looking deep inside the C++ code of wkHTMLtoPDF, I can only offer a _guess_ that it would be either pt (points) or perhaps px (pixels). Have you tried setting this option (fairly large) on a document with a range of font sizes, and see what the effect seems to be (by trial and error)? It shouldn't take more than an hour or two of work.

**Comment by zardilior** [comment]
The problem is that exactly in order to hit the exact unit conversion, which is needed sometimes it takes more or less an hour or two every time. Just because of that omission

**Comment by PhilterPaper** [comment]
I was thinking in terms of making a page with 18pt text, 17pt, 16pt... down to 4pt and seeing how it looked, and then running with the minimum-font-size=10 and seeing which ones changed (were larger). Does that not work? Does it all in one shot, if it does.

**Comment by zardilior** [comment]
@PhilterPaper  you are right, still once we do that we should write it down somewhere in the docs

**Comment by PhilterPaper** [comment]
I just tried it out, for type sizes ranging from 18pt down to 4pt. Assuming _those_ are correctly rendered, a minimum-font-size of 10 seems to affect only 7pt and smaller, 12 affects 8pt and smaller, and 15 affects 10pt and smaller. A point should be an absolute size, so if this is px (pixels), that could vary by dpi. If I also give --dpi, all the output is affected, but the size cutoff _seems_ to be unaffected.

In other words, it's still not absolutely clear what the units are, and whether they're absolute (pt, mm, etc.) or depend on resolution or even the typeface chosen. Someone still needs to look in the code and see if there are any hints there. It might even be hidden in WebKit somewhere!

[minfs.html.txt](https://github.com/wkhtmltopdf/wkhtmltopdf/files/3272552/minfs.html.txt)

**Comment by zardilior** [comment]
Will try to check code later

**Comment by zardilior** [comment]
I0ve checked the code and saw this, 

ws->setFontSize(QWebSettings::MinimumFontSize, s.minimumFontSize);

After some research it us part of the qt library components, and the unit is... still a mystery

**Comment by PhilterPaper** [comment]
If Opera and Konqueror use QT, then http://www.mit.edu/afs.new/sipb/project/opera/OldFiles/distrib/v9.10-solaris/opera-9.10-20061214.1-static-qt-sol8-sparc-en-521/locale/english.lng suggests that the units is pixels.

**Comment by zardilior** [comment]
That is truly bad, because it makes it so much harder specially because then you should configure it calculated on the dpi :(
